sandbox/apparmor/profile.go

```diff
@@ -287,6 +304,14 @@ var SnapConfineDistroProfilePath = func() string {
 	return ""
 }
 
+var loadHomedirs = func() ([]string, error) {
+	ssp, err := sysparams.Open("")
+	if err != nil {
+		return nil, err
+	}
+	return strutil.CommaSeparatedList(ssp.Homedirs), nil
+}
```

Comment:
this functions ends up always mocked so it is never exercised, it needs a dedicated test

Reply:
Really good point about this code not being hit! I don't think function deserves it's own dedicated test due to the nature of the function. Instead what I've done is make sure this code is excercised by the happy case `TestSetupSnapConfineSnippetsHomedirs` where it loads the actual system-params instead of mocking this function in all the test cases.